src/components/nav/TopBar.jsx

```diff
@@ -32,6 +32,14 @@ const TopBar = () => {
           link: path,
         };
       });
+    const checkIntInCrumb = () => {
+      breadcrumbsNodes.forEach((item) => {
+        if (!Number.isNaN(parseInt(item.name, 10))) {
+          item.name = <LoadingIcon icon="three-dots" className="w-8 h-8" />;
+        }
+      });
+    };
+    checkIntInCrumb();
```

Comment:
1. You did an extra iteration over the array that can be avoided
2. The icon must be added not to the data that gets into redux, but to the stage of rendering the react components
3. Why did you create the function?